### DISCUSSION TITLE
v1.33

### DISCUSSION THREAD

**Discussion by mikecao** [umami-software/umami#1228] [General]
## Important
This release will use Prisma migrations for the first time. This will help up make schema changes quickly and easily in the future in order to support new upcoming features. A new script called `check-db` has been added and will run before starting the app. It verifies that your database connection is defined and working and that your schema is up to date. The script will inform you what commands you need to run to get your database up to date. **You need to run the given command manually.** The message will looks something like this:

```
✗ Database is out of date.

You need to update your database by running:

yarn prisma migrate resolve --applied "20210320112717_init"
```

Also as a result of this change, the installation process will be much easier for new installation. There is no longer a manual database step to set up the tables. It will all be handled through Prisma. Additionally, defining the `HASH_SALT` environment variable is no longer required, but optional if you still want to use it.

## Features
- New language: Galacian! 💯 
- Added `IGNORE_HOSTNAME` environment variable. This will do a DNS lookup on a hostname and the resulting IP address will be ignored. This can be a comma delimited list of hostnames. 
- Added `DISABLE_UPDATES` environment variable which disables the new version check.

## Updates
- Start using Prisma database migrations.
- Removed `HASH_SALT` requirement.
- Language translation updates
- Updated Prisma to `v3.15.2`

## Fixes
- Fixed display of encoded characters
- Moved `DISABLE_LOGIN` check from middleware to build step since it has been causing issues on Vercel.
- Removed `password` field from accounts query

A huge thanks to all the contributors on this release! @anbraten @userdehghani @bilguun0203 @valtlfelipe @saosangmo @Maxime-J 

<hr /><em>This discussion was created from the release <a href='https://github.com/mikecao/umami/releases/tag/v1.33'>v1.33</a>.</em>

**Comment by ajfriesen** [comment]
Do we need to change behaviour for the docker container?

I am getting the latest 1.x release every day with watchtower (IIRC).

**Reply by saschwarz** [reply]
You might want to follow this issue https://github.com/mikecao/umami/issues/1232 Migrations currently need to be manually applied. But, there is a proposal to optionally apply them automatically.

**Comment by mikecao** [comment]
@ajfriesen If there are database changes, the start script will check and ask your to run a command to update your database if needed. Otherwise it will be just like before.

**Comment by mcnaveen** [comment]
Thanks for the update 🎉

**Comment by meirroth** [comment]
@mikecao Hi,
I'm running umami on Railway. Do I need to run any special commands before upgrading?

**Comment by mikecao** [comment]
@meirroth No, same upgrade process as usual, as long as you get the latest `v1.33.1`

**Reply by meirroth** [reply]
Thank you!!

**Comment by tairosonloa** [comment]
Hi @mikecao! I was stuck a few versions back because for me these were breaking changes that need to address manually and didn't have time until today.

Now, I'll update my umami instance (on vercel) and want to ask you. I have a very old instance (3 years using umami!). Can I safely remove the `HASH_SALT` env variable? Will it affect my existing data in my database or get corrupted somehow? What could be the possible side effects?

Thank you very much!

**Reply by mikecao** [reply]
You can remove it if you want but it doesn't hurt to leave it. Worse case you'll be logged out and have to log back in. But your data will be fine.
